### PR TITLE
PowerShell Core on Windows

### PR DESCRIPTION
**Pull request by splatteredbits**

### CODE REVIEW

File: Test/GitHubRelease.Tests.ps1 [hunk 1]
```diff
@@ -37,7 +44,9 @@ function GivenAssetUpdated
         $To
     )
 
-    Mock -CommandName 'Invoke-RestMethod' -ModuleName 'Whiskey' -ParameterFilter ([scriptblock]::Create("`$Uri -like '$To'"))
+    Mock -CommandName 'Invoke-RestMethod' `
+         -ModuleName 'Whiskey' `
+         -ParameterFilter ([scriptblock]::Create("`$Uri -like '$To'"))
```
[the remark this answers is not in the document]
Comment (splatteredbits) [reply Test/GitHubRelease.Tests.ps1]: This is the only way the scope of the `$Uri` and `$To` correct so their values are expected. 🤷‍♀️

File: Test/GitHubRelease.Tests.ps1 [hunk 2]
```diff
@@ -169,11 +184,31 @@ function ThenRequest
...
                 Write-Debug ('Body  expected  {0}' -f $WithBody)
                 Write-Debug ('      actual    {0}' -f $Body)
-                $Uri -eq [uri]$ToUri -and $Body -eq $WithBody 
+                $bodyEqual = $Body -eq $WithBody 
+                $WithBody = $WithBody | ConvertFrom-Json
+                $expectedProps = $WithBody | Get-Member -MemberType NoteProperty | Select-Object -ExpandProperty 'Name' | Sort-Object
+                $Body = [pscustomobject]@{ }
+                if( $Body )
+                {
+                    $Body | ConvertFrom-Json
```
[the remark this answers is not in the document]
Comment (splatteredbits) [reply Test/GitHubRelease.Tests.ps1]: Fixed.

File: Whiskey/Tasks/Publish-WhiskeyNuGetPackage.ps1 [hunk 3]
```diff
@@ -97,17 +97,29 @@ Use the `Add-WhiskeyApiKey` function to add the API key to the build.
...
         }
-        catch [Net.WebException]
+        catch
         {
-            $response = [Net.HttpWebResponse]([Net.WebException]$_.Exception).Response
+            # Invoke-WebRequest throws differnt types of errors in Windows PowerShell and PowerShell Core. Handle the case where a non-HTTP exception occurs.
+            if( -not ($_.Exception | Get-Member 'Response') )
+            {
+                Write-Error -ErrorRecord $_
+                Stop-WhiskeyTask -TaskContext $TaskContext -Message ('Unknown failure checking if {0} {1} package already exists at {2}. {3}' -f  $packageName,$packageVersion,$packageUri,$_)
```
[the remark this answers is not in the document]
Comment (splatteredbits) [reply Whiskey/Tasks/Publish-WhiskeyNuGetPackage.ps1]: It's a strange situation: `Invoke-WebRequest` not throwing an expected exception. I put the `Write-Error` there to see more error information than just the error message, which is what shows up in the `Stop-WhiskeyTask` message.

**Comment by splatteredbits** [comment]
@DecoyJoe Ready for next review.

**Comment by DecoyJoe** [comment]
@splatteredbits There's failing tests for the `GitHubRelease` task.

**Comment by splatteredbits** [comment]
Just fixed. They a

> @splatteredbits There's failing tests for the `GitHubRelease` task.

Just fixed. They were only failing under PowerShell Core. Ready for another review.

**Pull request merged**